lgc/patch/PatchResourceCollect.cpp

```diff
@@ -1690,20 +1693,34 @@ void PatchResourceCollect::matchGenericInOut() {
   LLPC_OUTS("// LLPC location input/output mapping results (" << getShaderStageAbbreviation(m_shaderStage)
                                                               << " shader)\n\n");
   unsigned nextMapLoc = 0;
-  if (!inLocMap.empty()) {
+  if (!inLocInfoMap.empty()) {
     assert(inOutUsage.inputMapLocCount == 0);
-    for (auto &locMap : inLocMap) {
-      assert(locMap.second == InvalidValue || m_pipelineState->isUnlinked());
+    // Update the value of inLocInfoMap for non pack case
+    // Note: The value of inLocInfoMap should be filled in packing code path
+    for (auto &locInfoMap : inLocInfoMap) {
+      InOutLocationInfo origLocInfo = {};
+      origLocInfo.u16All = locInfoMap.first;
+      InOutLocationInfo newLocInfo = {};
+
       // NOTE: For vertex shader, the input location mapping is actually trivial.
-      locMap.second = m_shaderStage == ShaderStageVertex ? locMap.first : nextMapLoc++;
-      inOutUsage.inputMapLocCount = std::max(inOutUsage.inputMapLocCount, locMap.second + 1);
-      LLPC_OUTS("(" << getShaderStageAbbreviation(m_shaderStage) << ") Input:  loc = " << locMap.first
-                    << "  =>  Mapped = " << locMap.second << "\n");
+      if (m_shaderStage == ShaderStageVertex) {
+        locInfoMap.second = origLocInfo.u16All;
+        newLocInfo.u16All = origLocInfo.u16All;
+      } else if (locInfoMap.second == InvalidValue) {
+        newLocInfo.location = nextMapLoc++;
+        locInfoMap.second = newLocInfo.u16All;
+      } else {
+        newLocInfo.u16All = locInfoMap.second;
```

Comment:
Could you leave a comment or assert about when this case is expected? Based on the assert above I think you could add assert(m_pipelineState->isUnlinked()) here

Reply:
Good point. It reminds that the input/output packing functionallity should be disabled for un-linked case. Un-linked case needs to keep all locations, while the unused locations need to be removed to guarantee the packing correctness. So I add `!m_unlink` condition to switch packing on or off. I add a condition `m_pipelineState->isUnlinked()` to set `locInfoMap.second` for non-vs input.